ksqldb-udf/src/main/java/io/confluent/ksql/function/FunctionCategory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package io.confluent.ksql.function;
+
+public enum FunctionCategory {
```

Comment:
it might be better to have these just be strings so that users can categorize their own into whatever categories they want

Reply:
i did wonder about that but decided i'd rather be restrictive so that we don't accidentally introduce a plethora of too-similar ones into the codebase over time. Perhaps a good compromise would be to have a "CUSTOM" enum value ? Maybe even set that as the default value so that if someone doesn't set a value for this in their own UDF then it just gets set correctly for them ? WDYT ?

Reply:
I don't think that should really be a concern if we group all of the values in a constant file just like we have here as an enum - people follow examples that are already there, so they'll tend to look in that file before introducing new ones. In the worst case, we'll have some unnecessary categories - but it would be compatible to change those and clean them up. 

My preference is to avoid enums in public APIs because they're difficult to extend and impossible to remove.

Reply:
gotcha, i'll change that then. 
Just for my own edification, why is it harder to extend or remove enum constants in an API than static strings ? i would naively suppose that removing existing things always has a similar backwards-compatibility problem, however they are defined ? And by "extending" here, do you mean adding new values ?

Reply:
> why is it harder to extend or remove enum constants in an API than static strings
A few reasons. One is logistically if external repos depend on a removed enum they can't just "add it in" and keep going (which you can if you depended on a removed string constant). Second is a little bit more philosophical, you shouldn't remove an enum because it changes the ordinals of everything after it - which I've been bit by before (though unlikely it'll fall into play here). I think JSON schema might actually fail compatibility checks if you removed an enum from a schema (not that we use json schema yet...)

> And by "extending" here, do you mean adding new values ?
yeah

Reply:
got it, that makes sense. thanks for taking the time to share!